src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1

```diff
@@ -401,21 +401,119 @@ class Organization: ADOSVTBase
+                        $display= $ExtensionListWithCriticalScopes|Format-Table -Property  @{name="ExtensionName";expression={$_.extensionName}},@{name="Scope";expression={$_.scopes}} | Out-String
+                        $controlResult.AddMessage($display) 
+
+                        # Avoid extensions  with 'DevTest', 'Demo', 'Preview', 'Deprecated' in names
+                        $ExtensionListWithInvalidExtensionNames=@()
+                        $InvalidExtensionNames=$this.ControlSettings.Organization.InvalidExtensionNames;
+                        for($i=0;$i -lt $extensionList.count;$i++)
+                        {
+                            for($j=0;$j -lt $InvalidExtensionNames.Count;$j++)
+                            {
+                                if($extensionList[$i] -match $InvalidExtensionNames[$j])
+                                {
+                                    [System.Array] $ExtensionListWithInvalidExtensionNames = $extensionList[$i]
+                                }
+                            }
+                        }                        
+                        if($ExtensionListWithInvalidExtensionNames.count -gt 0)
+                        {
+                        $controlResult.AddMessage("`nExtensions having name as of test environment : "+ $ExtensionListWithInvalidExtensionNames.count)
+                        $controlResult.AddMessage($ExtensionListWithInvalidExtensionNames) 
+                        }
+                        else {
+                            $controlResult.AddMessage("`nExtensions having name as of test environment : 0")
+                        }
+                        
+                        # Display extensions with Top Publishers, extensions that are private and Nonprod extensions
+                        $topPublisherExt=@()
+                        $privateExtensions=@()
+                        $nonProdExtensions=@()
+                        $extensionList | ForEach-Object {
+                            $url="https://marketplace.visualstudio.com/_apis/public/gallery/extensionquery?api-version=6.1-preview.1"
+                            $inputbody = "{
+                                'assetTypes': null,
+                                'filters': [
+                                    {
+                                        'criteria': [
+                                            {
+                                                'filterType': 7,
+                                                'value': '$($_.publisherId).$($_.extensionId)'
+                                            }
+                                        ]                                
+                                    }
+                                ]
+                            }" | ConvertFrom-Json | ConvertTo-json -Depth 10
```

Comment:
is the convulation from json to PSObject to json again reqd?

Reply:
removed as convolution was not required